### PR TITLE
Restructure documentation. Now the documentation has installation, administration, usage, development, contributing the 5 main parts

### PR DESCRIPTION
**Pull request by lunny**
- **Installation**: includes how to install Gitea and related other tools, also includes upgrade Gitea
- **Administration**: includes how to configure Gitea, customize Gitea and manage Gitea instance out of Gitea admin UI
- **Usage**: includes how to use Gitea's functionalities. A sub documentation is about packages, in future we could also include CI/CD and others.
- **Development**: includes how to integrate with Gitea's API, how to develop new features within Gitea
- **Contributing**: includes how to contribute code to Gitea repositories. 

After this is merged, I think we can have a sub-documentation of `Usage` part named `Actions` to describe how to use  Gitea actions

### CODE REVIEW

File: docs/Makefile
Comment (yardenshoham) [inline docs/Makefile:1]: Hugo version mismatch here:
https://github.com/go-gitea/gitea/blob/d4f35bd681af0632da988e15306f330e020422b2/Makefile#L820

**Review by yardenshoham — CHANGES_REQUESTED** [review]

File: docs/Makefile
Comment (lunny) [reply docs/Makefile:1]: Done.

**Review by yardenshoham — COMMENTED** [review]
image
`Comparison` seems out of place

**Comment by delvh** [comment]
Hmm, isn't `Installation` and `Administer` basically the same thing?
Could we perhaps combine them under `Setup`?

**Comment by lunny** [comment]
> Hmm, isn't `Installation` and `Administer` basically the same thing? Could we perhaps combine them under `Setup`?

I think you are right. But I move them as standalone documentation according to most people's habits.

File: Makefile [hunk 1]
```diff
@@ -817,7 +817,7 @@ release-docs: | $(DIST_DIRS) docs
 .PHONY: docs
 docs:
 	@hash hugo > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
-		curl -sL https://github.com/gohugoio/hugo/releases/download/v0.74.3/hugo_0.74.3_Linux-64bit.tar.gz | tar zxf - -C /tmp && mv /tmp/hugo /usr/bin/hugo && chmod +x /usr/bin/hugo; \
+		curl -sL https://github.com/gohugoio/hugo/releases/download/v0.111.3/hugo_0.111.3_Linux-64bit.tar.gz | tar zxf - -C /tmp && mv /tmp/hugo /usr/bin/hugo && chmod +x /usr/bin/hugo; \
```
Comment (delvh) [inline Makefile]: Wouldn't it be better to insert the version automatically?
I.e. `/download/v$(HUGO_VERSION)`?
Would of course require a new variable…

**Comment by lunny** [comment]
> image `Comparison` seems out of place

Looks like we need to change the theme about the style.

**Comment by wxiaoguang** [comment]
I think separate Installation from Administer is more friendly for new users.

File: docs/content/doc/packages/rubygems.en-us.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ https://gitea.example.com/api/packages/{owner}/rubygems: Bearer {token}
 | Parameter     | Description |
 | ------------- | ----------- |
 | `owner`       | The owner of the package. |
-| `token`       | Your [personal access token]({{< relref "doc/developers/api-usage.en-us.md#authentication" >}}). |
+| `token`       | Your [personal access token]({{< relref "doc/develop/api-usage.en-us.md#authentication" >}}). |
```
Comment (delvh) [inline docs/content/doc/packages/rubygems.en-us.md]: Hmm…
`PATs` only in `develop`?
Sounds like a future PR to move that to `usage` where it belongs…

File: docs/content/doc/pacakges.md
Comment (delvh) [inline docs/content/doc/pacakges.md:1]: Typo in the filename.

**Comment by delvh** [comment]
By the way, should we backport this PR to 1.19?
As we are moving towards versioned docs, it would also be nice there…

**Comment by wxiaoguang** [comment]
And I also have a (similar) question, what's the progress of Versioned Document? Does this change work with Versioned Document (how does it work with 1.19)?

File: docs/content/doc/packages/rubygems.en-us.md (hunk 2 above)
Comment (lunny) [reply docs/content/doc/packages/rubygems.en-us.md]: Leave it until we can merge swagger into here.

**Comment by lunny** [comment]
> And I also have a (similar) question, what's the progress of Versioned Document? Does this change work with Versioned Document (how does it work with 1.19)?

The versioned document repository is https://gitea.com/gitea/gitea-docusaurus . I think after this merged, it needs some adjustments.

**Comment by lunny** [comment]
Looks like there is no new feature merged after 1.19 branch created, so I think yes, we can backport this to 1.19 to make versioned documentation easier.

File: docs/content/doc/pacakges.md
Comment (lunny) [reply docs/content/doc/pacakges.md:1]: Done.

File: Makefile (hunk 1 above)
Comment (lunny) [reply Makefile]: Done.

File: Makefile [hunk 3]
```diff
@@ -817,7 +818,7 @@ release-docs: | $(DIST_DIRS) docs
 .PHONY: docs
 docs:
 	@hash hugo > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
-		curl -sL https://github.com/gohugoio/hugo/releases/download/v0.111.3/hugo_0.111.3_Linux-64bit.tar.gz | tar zxf - -C /tmp && mv /tmp/hugo /usr/bin/hugo && chmod +x /usr/bin/hugo; \
+		curl -sL https://github.com/gohugoio/hugo/releases/download/v$(HUGO_VERSION)/hugo_$(HUGO_VERSION)_Linux-64bit.tar.gz | tar zxf - -C /tmp && mv /tmp/hugo /usr/bin/hugo && chmod +x /usr/bin/hugo; \
```
Comment (yardenshoham) [inline Makefile:821]: This will break Gitpod. It's fine, I guess I will fix it after this is merged

**Review by delvh — APPROVED** [review]

**Comment by jolheiser** [comment]
We should find a spot for comparison before merging this.

Either by changing the theme, adding it to the index, or adding it to another section.

**Comment by delvh** [comment]
`Usage`?

File: docs/content/doc/usage/secrets.en-us.md [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 title: "Secrets"
-slug: "secrets/overview"
+slug: "usage/secrets/overview"
 draft: false
 toc: false
 menu:
   sidebar:
-    parent: "secrets"
+    parent: "usage"
     name: "Overview"
```
Comment (jolheiser) [inline docs/content/doc/usage/secrets.en-us.md]:
```suggestion
    name: "Secrets"
```

File: docs/content/doc/usage/secrets.en-us.md [hunk 5]
```diff
@@ -1,15 +1,15 @@
 ---
 date: "2022-12-19T21:26:00+08:00"
 title: "Secrets"
-slug: "secrets/overview"
+slug: "usage/secrets/overview"
```
Comment (jolheiser) [inline docs/content/doc/usage/secrets.en-us.md]:
```suggestion
slug: "usage/secrets"
```

**Review by jolheiser — CHANGES_REQUESTED** [review]
As well, I think "Help" should be moved back to the top imo

**Comment by delvh** [comment]
True, `Help` would also be the appropriate place for a FAQ…

**Review by yardenshoham — COMMENTED** [review]
Rename? Some options:

```
Install
Administer
Use
Develop
Contribute
```
or
```
Installation
Administration
Usage
Development
Contribution
```

File: Makefile (hunk 3 above)
Comment (yardenshoham) [reply Makefile:821]: Actually, this variable isn't used anywhere else so I'd rather just keeping the version hardcoded here

**Comment by lunny** [comment]
> Rename? Some options:
> 
> 
> 
> ```
> 
> Install
> 
> Administer
> 
> Use
> 
> Develop
> 
> Contribute
> 
> ```
> 
> or
> 
> ```
> 
> Installation
> 
> Administration
> 
> Usage
> 
> Development
> 
> Contribution
> 
> ```

I prefer first

**Comment by lafriks** [comment]
> Rename? Some options:
> 
> ```
> Install
> Administer
> Use
> Develop
> Contribute
> ```
> or
> ```
> Installation
> Administration
> Usage
> Development
> Contribution
> ```

I would prefer second more, it's more formal and more common in documentation

**Comment by lunny** [comment]
I think it should be Contributing but Contibution

**Comment by yardenshoham** [comment]
I agree, let's do:
```
Installation
Administration
Usage
Development
Contributing
```

File: Makefile (hunk 3 above)
Comment (lunny) [reply Makefile:821]: Since I have changed that and if it's equal. Maybe just leave it.

**Comment by lunny** [comment]
> As well, I think "Help" should be moved back to the top imo

I have a different thought about moving help in the last. The FAQ should be read after users read all the documentation but with no answer. And support chat channel should also be given when users cannot find the answer from the documentation. They are not encouraged to ask in the chatroom before they read the documentation carefully.

**Review by jolheiser — APPROVED** [review]

**Pull request merged**